Dockerfile

```diff
@@ -0,0 +1,16 @@
+FROM node:8
+
+# Create app directory
+WORKDIR /usr/src/app
+
+# Install app dependencies
+COPY package*.json ./
+
+RUN npm install
+
+# Bundle app source
+COPY . .
```

Comment:
This line copies everything in the source (current) directory to the working directory in the docker image. And, _.dockerignore_ will omit copying _./node_modules_ and _./npm-debug.log_. This approach essentially blacklists the files in the _.dockerignore_.

Instead of blacklisting files, I find it better to whitelist the files that you need. That is, only copy over the files necessary to run the app. This ensures that your docker image stays as small as possible and that you're only copying over the files that you need.

So, you wouldn't need a _.dockerignore_ file, just specify all the files to be copied over in the working directory. This also helps another contributor to understand exactly which files are needed to run the application.